### PR TITLE
Blog article on Keylime IMA

### PR DESCRIPTION
**Pull request by lukehinds**

### CODE REVIEW

File: _posts/2019-04-02-running-IMA-on-keylime.markdown [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+
+```
+# tpm2_getrandom 8
+0x0A 0xBB 0xEB 0xB7 0x82 0x40 0xC1 0x08
+```
+
+On any machine (this could be an unplugged machine for extra security) run the `create_whitelist.sh` script as follows:
+
+```
+# ./create_whitelist.sh ~/whitelist.txt initramfs sha1sum
```
Comment (lukehinds) [inline _posts/2019-04-02-running-IMA-on-keylime.markdown]: opps, I think we chnage this now @jetwhiz

File: _posts/2019-04-02-running-IMA-on-keylime.markdown [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+# MEASUREMENTS
+measure func=BPRM_CHECK
+measure func=FILE_MMAP mask=MAY_EXEC
+measure func=MODULE_CHECK uid=0
+```
+
+This default policy measures all executables in `bprm_check`, all files mmapped executable in `file_mmap`, and all files
+open for read by `root` in `do_filp_open`.
+
+It is of course possible to create your own IMA policy, along with LSM specific definitions for `SELinux` or `Smack`.
```
Comment (jetwhiz) [inline _posts/2019-04-02-running-IMA-on-keylime.markdown:63]: Not sure if we want to include this explicitly, but one example of using SELinux here is to have IMA watch all files in /etc/: 
`measure func=FILE_CHECK mask=MAY_READ obj_type=etc_t`

File: _posts/2019-04-02-running-IMA-on-keylime.markdown (hunk 1 above)
Comment (jetwhiz) [reply _posts/2019-04-02-running-IMA-on-keylime.markdown]: Yeah, with https://github.com/keylime/python-keylime/pull/88, create_whitelist.sh will only take up to two arguments (e.g., 'whitelist.txt' and 'sha1sum').

File: _posts/2019-04-02-running-IMA-on-keylime.markdown [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+
+IMA (also known as the kernel integrity subsystem) provides a means of detecting whether files have been accidentally or maliciously altered or executed both remotely or locally.
+
+The following modules provide the integrity functions:
+
+* Collect – measure a file before it is accessed.
+* Store – add the measurement to a kernel resident list and, if a
+hardware Trusted Platform Module (TPM) is present, extend the IMA
+PCR
+* Attest – if present, use the TPM to sign the IMA PCR value, to
```
Comment (jetwhiz) [inline _posts/2019-04-02-running-IMA-on-keylime.markdown:23]: Note that IMA itself is not doing the attesting, but writes the TPM PCR values necessary to be able to attest runtime state

**Comment by jetwhiz** [comment]
This looks great @lukehinds, thanks for putting this together!  Just a few comments to look at.

File: _posts/2019-04-02-running-IMA-on-keylime.markdown (hunk 3 above)
Comment (lukehinds) [reply _posts/2019-04-02-running-IMA-on-keylime.markdown:23]: This is just a copy / paste from the IMA readme. Happy to change if you could give me that one sentence reworded.

File: _posts/2019-04-02-running-IMA-on-keylime.markdown (hunk 2 above)
Comment (lukehinds) [reply _posts/2019-04-02-running-IMA-on-keylime.markdown:63]: Nice! I added that

**Pull request merged**